### PR TITLE
#166265851 Get author/ Update author endpoints

### PR DESCRIPTION
**Pull request by ocranbillions**
### What does this PR do?
- This PR adds the functionalities to getAuthor and updateAuthor

#### Description of Task to be completed?
- define association between authors and books relations
- create route for get single author
- add controller for getting author
- return response to user
- write unit tests

- create route for update-author
- add validation middle were for inputs
- add controller for to perform the update
- return response to user
- write unit tests

#### How should this be manually tested?
- Make a get request to `api/v1/authors/n` where N is the id of the author being searched for
- To update, make a patch request to `api/v1/authors/n` where N is the id of the author to be updated

#### Any background context you want to provide?
- This Task was assigned to Kafilat and I. I did the getAuthor and updateAuthor features. For now, Kafilat is waiting for this PR to be merged so she can add the deteleAuthor and API documentation for all three endpoints.

#### What are the relevant pivotal tracker stories?
- [166265851](https://www.pivotaltracker.com/story/show/166265851)

#### Screenshots (if appropriate)
- GET AUTHOR
get

- UPDATE AUTHOR
update

#### Questions:
- N/A

### CODE REVIEW

**Review by Hadeneekeh — APPROVED** [review]

**Review by encodedBicoding — APPROVED** [review]

File: server/__test__/mock_data/authors.mock.js [hunk 1]
```diff
@@ -16,6 +16,18 @@ const author = {
...
+
+  updateAuthor1: {
+    firstName: 'Kelvin',
+    middleName: 'Nick',
+    lastName: 'Simon'
+  },
+
+  invalidUpdate: {
+    firstName: 'Kelvin',
+    middleName: '419',
```
Comment (chuxmykel) [inline server/__test__/mock_data/authors.mock.js:28]: Lmao. Really?

File: server/db/migrations/20190605110045-create-book.js [hunk 2]
```diff
@@ -21,8 +21,13 @@ export function up(queryInterface, Sequelize) {
       type: Sequelize.NUMERIC
     },
     authorID: {
-      allowNull: true,
-      type: Sequelize.INTEGER
+      allowNull: false,
```
Comment (chuxmykel) [inline server/db/migrations/20190605110045-create-book.js:24]: I thought this was set to null for an reason

Comment (ocranbillions) [reply server/db/migrations/20190605110045-create-book.js:24]: I can change it back if it breaks anything. The reason I set it to not allow false is because then @PelumiAlesh has a validation check on it that states `"AuthorID can not be left empty"`

File: server/db/models/book.js [hunk 3]
```diff
@@ -13,10 +13,6 @@ export default (sequelize, DataTypes) => {
       allowNull: false,
       type: DataTypes.NUMERIC
     },
-    authorID: {
```
Comment (emp-daisy) [inline server/db/models/book.js:16]: WHy was this removed?

**Comment by emp-daisy** [comment]
I didn't see the documentation updated for the new changes

File: server/db/models/book.js (hunk 3 above)
Comment (ocranbillions) [reply server/db/models/book.js:16]: I removed it there, to re-defined it in the book-migration file line 23. I read from the link below that if you have a foreignkey defined in both the model and the migration files. sequelize will throw an error. So it only has to be defined in the migration file of that model. 
https://www.duringthedrive.com/2017/05/06/models-migrations-sequelize-node/

Comment (emp-daisy) [reply server/db/models/book.js:16]: I see
